thirdparty/github.go

```diff
@@ -497,82 +474,68 @@ func GithubAuthenticate(code, clientId, clientSecret string) (githubResponse *Gi
 	return
 }
 
-// GetGithubUser does a GET from GitHub for the user, email, and organizations information and
-// returns the GithubLoginUser and its associated GithubOrganizations after authentication
-func GetGithubUser(token string) (githubUser *GithubLoginUser, githubOrganizations []GithubOrganization, err error) {
-	userUrl := fmt.Sprintf("%v/user", GithubAPIBase)
-	orgUrl := fmt.Sprintf("%v/user/orgs", GithubAPIBase)
-	t := fmt.Sprintf("token %v", token)
-	// get the user
-	resp, err := tryGithubGet(t, userUrl)
-	if resp != nil {
-		defer resp.Body.Close()
-	}
-	if err != nil {
-		return nil, nil, errors.WithStack(err)
-	}
-	respBody, err := ioutil.ReadAll(resp.Body)
+// GetGithubUser fetches a github user associated with an oauth token, and
+// if requiredOrg is specified, checks that it belongs to that org.
+// Returns user object, if it was a member of the specified org (or false if not specified),
+// and error
+func GetGithubUser(token string, requiredOrg string) (*GithubLoginUser, bool, error) {
+	httpClient, err := util.GetHttpClientForOauth2(fmt.Sprintf("token %s", token))
 	if err != nil {
-		return nil, nil, ResponseReadError{err.Error()}
-	}
-
-	grip.Debugf("Github API response: %s. %d bytes", resp.Status, len(respBody))
-
-	if err = json.Unmarshal(respBody, &githubUser); err != nil {
-		return nil, nil, APIUnmarshalError{string(respBody), err.Error()}
+		return nil, false, errors.Wrap(err, "can't fetch data from github")
 	}
+	defer util.PutHttpClientForOauth2(httpClient)
+	client := github.NewClient(httpClient)
 
-	// get the user's organizations
-	resp, err = tryGithubGet(t, orgUrl)
-	if resp != nil {
-		defer resp.Body.Close()
-	}
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "Could not get user from token")
-	}
-	respBody, err = ioutil.ReadAll(resp.Body)
+	user, resp, err := client.Users.Get(context.TODO(), "")
```

Comment:
is it reasonable based on where this is called to pass the context in?

Reply:
no callers of these methods have contexts, so they'd be using context.TODO(). That said, i'm still going to go and amend these with contexts so that timeouts can be used.